### PR TITLE
Prepare 4.1.7 release

### PR DESCRIPTION
**Pull request by joshmgross**
This release includes the changes from #1739, #1697, #1774, and #1776

### CODE REVIEW

**Review by cory-miller — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## v4.1.7
+* Bump the minor-npm-dependencies group across 1 directory with 4 updates by @dependabot in https://github.com/actions/checkout/pull/1739
+* Bump actions/checkout from 3 to 4 by @dependabot in https://github.com/actions/checkout/pull/1697
+* Check out other refs/* by commit by @orhantoy in https://github.com/actions/checkout/pull/1774
```
Comment (jww3) [inline CHANGELOG.md:6]:
```suggestion
## v4.1.7
* Bump the minor-npm-dependencies group across 1 directory with 4 updates by @dependabot in https://github.com/actions/checkout/pull/1739
* Bump actions/checkout from 3 to 4 by @dependabot in https://github.com/actions/checkout/pull/1697
* Check out other refs/* by commit by @orhantoy in https://github.com/actions/checkout/pull/1774
* Pin actions/checkout's own workflows to a known, good, stable version by @jww3 in https://github.com/actions/checkout/pull/1776
```

Comment (jww3) [reply CHANGELOG.md:6]: We probably want to merge https://github.com/actions/checkout/pull/1776 before this PR.

**Review by jww3 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## v4.1.7
+* Bump the minor-npm-dependencies group across 1 directory with 4 updates by @dependabot in https://github.com/actions/checkout/pull/1739
+* Bump actions/checkout from 3 to 4 by @dependabot in https://github.com/actions/checkout/pull/1697
```
Comment (AngelicaOvesdotter) [inline CHANGELOG.md:5]: image

**Review by xXxrAinSonic — COMMENTED** [review]
Ok

**Review by Vue-Pu — APPROVED** [review]

**Pull request merged**